app/controllers/users_controller.rb

```diff
@@ -1803,29 +1803,43 @@ def user_menu_messages
       raise Discourse::InvalidAccess.new("personal messages are disabled.")
     end
 
-    message_notifications = Notification.unread_type(
-      current_user,
-      Notification.types[:private_message],
-      USER_MENU_LIST_LIMIT
-    )
+    unread_notifications = Notification
+      .unread_types(
+        current_user,
+        [Notification.types[:private_message], Notification.types[:group_message_summary]],
+        USER_MENU_LIST_LIMIT
+      )
+      .prioritized(USER_MENU_LIST_LIMIT)
```

Comment:
The two scopes here when used together feels awkward where the same limit is passed in twice. I would considering refactoring scope so that a limit can be apply once only.